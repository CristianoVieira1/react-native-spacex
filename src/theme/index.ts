import { Dimensions } from "react-native";

const width = Dimensions.get("window").width;
const height = Dimensions.get("window").height;

export default {
  device: {
    width: width,
    height: height,
  },
  border: {
    radius: "12px",
  },
  fonts: {
    regular: "PublicSans_400Regular",
    medium: "PublicSans_500Medium",
    bold: "PublicSans_700Bold",
    sizes: {
      xxsmall: "10px",
      xsmall: "10px",
      small: "14px",
      medium: "18px",
      large: "24px",
      xlarge: "30px",
      xxlarge: "39px",
      xxxlarge: "64px",
    },
  },
  colors: {
    white: "#FFFFFF",
    black: "#000000",
    orange: "#FF8200",
    border: "#C5CADA",
    red: "#FF0000",
    gray: "#808080",
    lightGray: "#9CA3AF",
    opacityBlack: "#575F69",
    buttonActive: "#007bff",
    darkGray: "#1F2937",
  },
  shadow: {
    shadowOpacity: 0.5,
    shadowOffset: { width: 5, height: 3 },
    shadowRadius: 2,
    elevation: 5,
    shadowColor: "#00000066",
  },
  spacings: {
    xxsmall: "6px",
    xsmall: "12px",
    small: "14px",
    medium: "16px",
    large: "18px",
    xlarge: "24px",
    xxlarge: "34px",
    xxxlarge: "42px",
  },
};
