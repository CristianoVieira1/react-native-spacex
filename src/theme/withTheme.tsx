import React from "react";
import { ThemeProvider } from "styled-components/native";
import Theme from ".";

const withTheme = (component: React.ReactNode) => (
  <ThemeProvider theme={Theme}>{component}</ThemeProvider>
);

export { withTheme };
