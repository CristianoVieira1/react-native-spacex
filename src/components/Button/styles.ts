import styled from "styled-components/native";
import theme from "../../theme";

export type TypeButton =
  | "accept"
  | "disable"
  | "link"
  | "youtube"
  | "auxiliary";

export type SizeButton = "small" | "medium" | "large";
interface ButtonProps {
  type: TypeButton;
  size: SizeButton;
}

interface ButtonTitleProps {
  type: TypeButton;
}

const buttonColor = {
  accept: theme.colors.white,
  youtube: theme.colors.red,
  link: theme.colors.white,
  disable: theme.colors.lightGray,
  auxiliary: theme.colors.white,
};

const buttonTitleColor = {
  accept: theme.colors.darkGray,
  youtube: theme.colors.white,
  link: theme.colors.darkGray,
  disable: theme.colors.lightGray,
  auxiliary: theme.colors.darkGray,
};

const sizeButton = {
  small: "30",
  medium: "70",
  large: "100",
};

export const Container = styled.TouchableOpacity<ButtonProps>`
  width: ${({ size }) => (size ? sizeButton[size] : 100)}%;
  background: ${({ type }) => buttonColor[type]};
  height: 40px;
  align-items: center;
  justify-content: center;
  border-radius: 8px;
  margin-bottom: 10px;
`;

export const TitleButton = styled.Text<ButtonTitleProps>`
  font-size: ${({ theme }) => theme.fonts.sizes.small};
  font-family: ${({ theme }) => theme.fonts.bold};
  color: ${({ type }) => buttonTitleColor[type]};
  text-decoration-color: ${({ theme }) => theme.colors.orange};
`;
