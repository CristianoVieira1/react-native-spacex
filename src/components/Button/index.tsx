import Load from "@components/Animations/Load";
import React, { memo } from "react";
import { TouchableOpacityProps } from "react-native";
import theme from "../../theme";
import * as S from "./styles";

interface IButtonProps extends TouchableOpacityProps {
  title: string;
  loading?: boolean;
  type: S.TypeButton;
  size?: S.SizeButton;
}

function Button({
  title,
  loading = false,
  type,
  size = "medium",
  ...rest
}: IButtonProps) {
  return (
    <S.Container
      {...rest}
      activeOpacity={0.7}
      type={type}
      size={size}
      style={type !== "link" && { ...theme.shadow }}
      testID="component-button"
    >
      {loading ? <Load /> : <S.TitleButton type={type}>{title}</S.TitleButton>}
    </S.Container>
  );
}
export default memo(Button);
