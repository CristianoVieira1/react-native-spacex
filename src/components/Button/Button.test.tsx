import { fireEvent, render } from "@testing-library/react-native";
import React from "react";
import Button from ".";
import { withTheme } from "../../theme/withTheme";

describe("Button Component", () => {
  it("render Load", () => {
    const renderComponent = render(
      withTheme(<Button title="teste" type="accept" onPress={() => {}} />)
    );
    expect(renderComponent.toJSON()).toMatchSnapshot();
  });

  it("Button onClick accept", () => {
    const buttonIconOnly = render(
      withTheme(
        <Button
          title="Teste"
          loading={false}
          size="small"
          type="accept"
          onPress={() => {
            return "teste";
          }}
        />
      )
    );

    const buttonClick = buttonIconOnly.getByTestId("component-button");
    fireEvent.press(buttonClick);
  });

  it("Button onClick youtube ", () => {
    const buttonIconOnly = render(
      withTheme(
        <Button
          title="Teste"
          loading={false}
          size="small"
          type="youtube"
          onPress={() => {
            return "teste";
          }}
        />
      )
    );

    const buttonClick = buttonIconOnly.getByTestId("component-button");
    fireEvent.press(buttonClick);
  });

  it("Button onClick link", () => {
    const buttonIconOnly = render(
      withTheme(
        <Button
          title="Teste"
          loading={false}
          size="small"
          type="link"
          onPress={() => {
            return "teste";
          }}
        />
      )
    );

    const buttonClick = buttonIconOnly.getByTestId("component-button");
    fireEvent.press(buttonClick);
  });
});
