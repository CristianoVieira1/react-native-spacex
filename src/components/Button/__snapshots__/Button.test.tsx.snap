// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component render Load 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 8,
      "elevation": 5,
      "height": 40,
      "justifyContent": "center",
      "marginBottom": 10,
      "opacity": 1,
      "shadowColor": "#00000066",
      "shadowOffset": {
        "height": 3,
        "width": 5,
      },
      "shadowOpacity": 0.5,
      "shadowRadius": 2,
      "width": "70%",
    }
  }
  testID="component-button"
>
  <Text
    style={
      [
        {
          "color": "#1F2937",
          "fontFamily": "PublicSans_700Bold",
          "fontSize": 14,
          "textDecorationColor": "#FF8200",
        },
      ]
    }
    type="accept"
  >
    teste
  </Text>
</View>
`;
